README.md
# NAME

Dist::Zilla::Plugin::Author::CSSON::GithubActions - Ease creation of common Github Actions workflows

<div>
    <p>
    <img src="https://img.shields.io/badge/perl-5.14+-blue.svg" alt="Requires Perl 5.14+" />
    <img src="https://img.shields.io/badge/coverage-84.6%25-orange.svg" alt="coverage 84.6%" />
    <a href="https://github.com/Csson/p5-Dist-Zilla-Plugin-Author-CSSON-GithubActions/actions?query=workflow%3Amakefile-test"><img src="https://img.shields.io/github/workflow/status/Csson/p5-Dist-Zilla-Plugin-Author-CSSON-GithubActions/makefile-test" alt="Build status at Github" /></a>
    </p>
</div>

# VERSION

Version 0.0107, released 2020-12-30.

# SYNOPSIS

In dist.ini:

    [Author::CSSON::GithubActions::Workflow::TestWithMakefile]
    ; set on.push.branches to an empty list
    clear_on_push_branches = 1

    ; set on.pull_request.branches to an empty list
    clear_on_pull_request_branches = 1

    ; add branches to on.push.branches
    on_pull_request_branches = 'this-branch'
    on_pull_request_branches = 'that-other-branch'

    ; add branches to on.pull_request.branches
    on_pull_request_branches = 'my-pr-branch'
    on_pull_request_branches = 'feature-branch'

    ; replace jobs.perl-job.strategy.matrix.os
    matrix_os = ubuntu-latest
    matrix_os = ubuntu-16.04

    ; replace jobs.perl-job.strategy.matrix.perl-version
    perl_version = 5.32
    perl_version = 5.24
    perl_version = 5.18

# STATUS

This plugin is very early in development. It might be released in a non-author namespace at a later stage.

# DESCRIPTION

This distribution is a framework for creating re-usable Github Actions workflows. GitHub's documentation about Actions is located at [http://docs.github.com/en/free-pro-team@latest/actions](http://docs.github.com/en/free-pro-team@latest/actions).

The core of it is the [Dist::Zilla::Role::Author::CSSON::GithubActions](https://metacpan.org/pod/Dist::Zilla::Role::Author::CSSON::GithubActions) role.

Included in this distribution is an example workflow, [Dist::Zilla::Plugin::Author::CSSON::GithubActions::Workflow::TestWithMakefile](https://metacpan.org/pod/Dist::Zilla::Plugin::Author::CSSON::GithubActions::Workflow::TestWithMakefile).

# SEE ALSO

- [Dist::Zilla::TravisCI](https://metacpan.org/pod/Dist::Zilla::TravisCI)
- [http://docs.github.com/en/free-pro-team@latest/actions](http://docs.github.com/en/free-pro-team@latest/actions)
- [https://perlmaven.com/setup-github-actions](https://perlmaven.com/setup-github-actions)

# SOURCE

[https://github.com/Csson/p5-Dist-Zilla-Plugin-Author-CSSON-GithubActions](https://github.com/Csson/p5-Dist-Zilla-Plugin-Author-CSSON-GithubActions)

# HOMEPAGE

[https://metacpan.org/release/Dist-Zilla-Plugin-Author-CSSON-GithubActions](https://metacpan.org/release/Dist-Zilla-Plugin-Author-CSSON-GithubActions)

# AUTHOR

Erik Carlsson <info@code301.com>

# COPYRIGHT AND LICENSE

This software is copyright (c) 2020 by Erik Carlsson.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for Dist-Zilla-Plugin-Author-CSSON-GithubActions

{{$NEXT}}
 [API Changes]

 [Bug Fixes]

 [Documentation]

 [Enhancements]

 [New Features]

 [Other Changes]

 [Requirements]

 [Stats & Dependency Changes]

 [Testing]

0.0107 2020-12-30T20:04:02Z UTC
 [Requirements]
 - Added missing prereq, fixes #2 (thanks SREZIC!)

 [Stats & Dependency Changes]
 - 5 files changed, 53 insertions(+), 13 deletions(-)
 - (run req) + Types::Path::Tiny 0.005
 - (run req) ~ perl 5.010000 --> 5.014000
 - (dev req) + Badge::Depot::Plugin::Githubactions (any)
 - (dev req) - Badge::Depot::Plugin::Cpantesters
 - (dev req) - Badge::Depot::Plugin::Kwalitee
 - (dev req) - Dist::Zilla::Plugin::Test::Kwalitee::Extra
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0327 --> 0.0328

0.0106 2020-12-26T15:26:00Z UTC
 [Bug Fixes]
 - Fix bugs in distribution_name, so that it looks in the correct share
   directory.

 [Stats & Dependency Changes]
 - 4 files changed, 13 insertions(+), 4 deletions(-)

0.0105 2020-12-26T13:07:13Z UTC
 [Bug Fixes]
 - Fix faulty package name

 [Stats & Dependency Changes]
 - 3 files changed, 7 insertions(+), 4 deletions(-)

0.0104 2020-12-26T12:56:52Z UTC
 [Other Changes]
 - Rewritten so that GithubActions is now a role, and each workflow class
   will be their own plugin. This makes it much easier to add custom
   parameters adapted for each workflow.

 [Stats & Dependency Changes]
 - 11 files changed, 374 insertions(+), 374 deletions(-)
 - (run req) - Module::Load

0.0103 2020-12-25T22:36:10Z UTC
 [Bug Fixes]
 - It was wrongfully listed as a prereq for itself, that has been fixed

 [Stats & Dependency Changes]
 - 4 files changed, 3 insertions(+), 4 deletions(-)
 - (run req) - Dist::Zilla::Plugin::Author::CSSON::GithubActions

0.0102 2020-12-25T20:57:32Z UTC
 [Bug Fixes]
 - Don't include the generated workflow file in distributions that uses
   this plugin

 [Stats & Dependency Changes]
 - 4 files changed, 24 insertions(+), 15 deletions(-)

0.0101 2020-12-25T19:49:39Z UTC
 [Bug Fixes]
 - Minor fixes

 [Stats & Dependency Changes]
 - 4 files changed, 9 insertions(+), 7 deletions(-)

0.0100 2020-12-25T18:19:22Z UTC
 - Initial release

 [Stats & Dependency Changes]
 - (run req) + Dist::Zilla::File::InMemory (any)
 - (run req) + Dist::Zilla::Plugin::Author::CSSON::GithubActions (any)
 - (run req) + Dist::Zilla::Role::FileGatherer (any)
 - (run req) + Dist::Zilla::Role::Plugin (any)
 - (run req) + File::ShareDir 1.102
 - (run req) + List::AllUtils (any)
 - (run req) + Module::Load 0.26
 - (run req) + Moose 2.1803
 - (run req) + Moose::Role (any)
 - (run req) + Path::Class::File (any)
 - (run req) + Path::Tiny 0.072
 - (run req) + Test::Exception (any)
 - (run req) + Try::Tiny 0.24
 - (run req) + Types::Standard 1.000006
 - (run req) + YAML::XS (any)
 - (run req) + namespace::autoclean 0.22
 - (run req) + perl 5.010000
 - (test req) + ExtUtils::MakeMaker (any)
 - (test req) + File::Spec (any)
 - (test req) + IO::Handle (any)
 - (test req) + IPC::Open3 (any)
 - (test req) + Path::Tiny 0.072
 - (test req) + Test::DZil (any)
 - (test req) + Test::Exception (any)
 - (test req) + Test::More 0.96
 - (test req) + YAML::XS (any)
 - (test rec) + CPAN::Meta 2.120900
 - (conf req) + ExtUtils::MakeMaker (any)
 - (conf req) + File::ShareDir::Install 0.06
 - (dev req) + Badge::Depot 0.0103
 - (dev req) + Badge::Depot::Plugin::Coverage (any)
 - (dev req) + Badge::Depot::Plugin::Cpantesters (any)
 - (dev req) + Badge::Depot::Plugin::Kwalitee (any)
 - (dev req) + Badge::Depot::Plugin::Perl (any)
 - (dev req) + Dist::Zilla 6.015
 - (dev req) + Dist::Zilla::Plugin::Authority 1.009
 - (dev req) + Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.009
 - (dev req) + Dist::Zilla::Plugin::ChangeStats::Dependencies::Git 0.0200
 - (dev req) + Dist::Zilla::Plugin::CheckChangesHasContent 0.011
 - (dev req) + Dist::Zilla::Plugin::Clean 0.07
 - (dev req) + Dist::Zilla::Plugin::ConfirmRelease (any)
 - (dev req) + Dist::Zilla::Plugin::CopyFilesFromBuild 0.170880
 - (dev req) + Dist::Zilla::Plugin::DistIller::MetaGeneratedBy (any)
 - (dev req) + Dist::Zilla::Plugin::ExecDir (any)
 - (dev req) + Dist::Zilla::Plugin::Git 2.046
 - (dev req) + Dist::Zilla::Plugin::Git::Check (any)
 - (dev req) + Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch 0.014
 - (dev req) + Dist::Zilla::Plugin::Git::Commit (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Contributors 0.035
 - (dev req) + Dist::Zilla::Plugin::Git::GatherDir (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Push (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Tag (any)
 - (dev req) + Dist::Zilla::Plugin::GithubMeta 0.54
 - (dev req) + Dist::Zilla::Plugin::InstallRelease 0.008
 - (dev req) + Dist::Zilla::Plugin::License (any)
 - (dev req) + Dist::Zilla::Plugin::MakeMaker (any)
 - (dev req) + Dist::Zilla::Plugin::Manifest (any)
 - (dev req) + Dist::Zilla::Plugin::ManifestSkip (any)
 - (dev req) + Dist::Zilla::Plugin::MetaConfig (any)
 - (dev req) + Dist::Zilla::Plugin::MetaJSON (any)
 - (dev req) + Dist::Zilla::Plugin::MetaNoIndex (any)
 - (dev req) + Dist::Zilla::Plugin::MetaProvides::Package 2.004003
 - (dev req) + Dist::Zilla::Plugin::MetaYAML (any)
 - (dev req) + Dist::Zilla::Plugin::NextRelease::Grouped 0.0200
 - (dev req) + Dist::Zilla::Plugin::PodSyntaxTests (any)
 - (dev req) + Dist::Zilla::Plugin::PodWeaver 4.006
 - (dev req) + Dist::Zilla::Plugin::Prereqs (any)
 - (dev req) + Dist::Zilla::Plugin::PromptIfStale 0.057
 - (dev req) + Dist::Zilla::Plugin::Readme (any)
 - (dev req) + Dist::Zilla::Plugin::ReadmeAnyFromPod 0.163250
 - (dev req) + Dist::Zilla::Plugin::RewriteVersion::Transitional 0.007
 - (dev req) + Dist::Zilla::Plugin::RunExtraTests 0.029
 - (dev req) + Dist::Zilla::Plugin::ShareDir (any)
 - (dev req) + Dist::Zilla::Plugin::Test::Compile 2.058
 - (dev req) + Dist::Zilla::Plugin::Test::EOF 0.0501
 - (dev req) + Dist::Zilla::Plugin::Test::EOL 0.18
 - (dev req) + Dist::Zilla::Plugin::Test::Kwalitee::Extra v0.2.1

Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.017.
use strict;
use warnings;

use 5.014000;

use ExtUtils::MakeMaker;

use File::ShareDir::Install;
$File::ShareDir::Install::INCLUDE_DOTFILES = 1;
$File::ShareDir::Install::INCLUDE_DOTDIRS = 1;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "Ease creation of common Github Actions workflows",
  "AUTHOR" => "Erik Carlsson <info\@code301.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::ShareDir::Install" => "0.06"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Author-CSSON-GithubActions",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.014000",
  "NAME" => "Dist::Zilla::Plugin::Author::CSSON::GithubActions",
  "PREREQ_PM" => {
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::Plugin" => 0,
    "File::ShareDir" => "1.102",
    "List::AllUtils" => 0,
    "Moose" => "2.1803",
    "Moose::Role" => 0,
    "Path::Class::File" => 0,
    "Path::Tiny" => "0.072",
    "Test::Exception" => 0,
    "Try::Tiny" => "0.24",
    "Types::Path::Tiny" => "0.005",
    "Types::Standard" => "1.000006",
    "YAML::XS" => 0,
    "namespace::autoclean" => "0.22"
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Path::Tiny" => "0.072",
    "Test::DZil" => 0,
    "Test::Exception" => 0,
    "Test::More" => "0.96",
    "YAML::XS" => 0
  },
  "VERSION" => "0.0108",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Role::Plugin" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::ShareDir" => "1.102",
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "List::AllUtils" => 0,
  "Moose" => "2.1803",
  "Moose::Role" => 0,
  "Path::Class::File" => 0,
  "Path::Tiny" => "0.072",
  "Test::DZil" => 0,
  "Test::Exception" => 0,
  "Test::More" => "0.96",
  "Try::Tiny" => "0.24",
  "Types::Path::Tiny" => "0.005",
  "Types::Standard" => "1.000006",
  "YAML::XS" => 0,
  "namespace::autoclean" => "0.22"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}
